lib/plugins/package/lib/zipService.js

```diff
@@ -131,53 +137,62 @@ function excludeNodeDevDependencies(servicePath) {
     });
 
     // filter out non node_modules file paths
-    const relevantFilePaths = _.filter(packageJsonFilePaths, (filePath) => {
+    const packageJsonPaths = _.filter(packageJsonFilePaths, (filePath) => {
       const isNodeModulesDir = !!filePath.match(/node_modules/);
       return !isNodeModulesDir;
     });
 
-    _.forEach(relevantFilePaths, (relevantFilePath) => {
+    return BbPromise.map(packageJsonPaths, (packageJsonPath) => {
       // the path where the package.json file lives
-      const fullPath = path.join(servicePath, relevantFilePath);
-      const rootDirPath = fullPath.replace(path.join(path.sep, 'package.json'), '');
+      const fullPath = path.join(servicePath, packageJsonPath);
+      const dirWithPackageJson = fullPath.replace(path.join(path.sep, 'package.json'), '');
 
-      process.chdir(rootDirPath);
+      const childProc = childProcess.spawn('npm',
+        ['ls', '--dev=true', '--parseable=true', '--silent'],
+        { cwd: dirWithPackageJson }
+      );
 
-      // TODO replace with package-manager independent directory traversal?!
-      const prodDependencies = childProcess
-        .execSync('npm ls --prod=true --parseable=true --silent')
-        .toString().trim();
+      let result = [];
 
-      const nodeModulesRegex = new RegExp(`${path.join('node_modules', path.sep)}.*`, 'g');
-      const prodDependencyPaths = prodDependencies.match(nodeModulesRegex);
-
-      let pathToDep = '';
-      // if the package.json file is not in the root of the service path
-      if (rootDirPath !== servicePath) {
-        // the path without the servicePath prepended
-        const relativeFilePath = rootDirPath.replace(path.join(servicePath, path.sep), '');
-        pathToDep = relativeFilePath ? `${relativeFilePath}/` : '';
-      }
+      return new BbPromise((resolve) => {
+        if (childProc.error) {
+          return resolve(result);
+        }
+
+        const chunks = [];
+
+        childProc.stdout.on('data', (chunk) => {
+          chunks.push(chunk);
+        });
 
-      const includePatterns = _.map(prodDependencyPaths, (depPath) =>
-        `${pathToDep}${depPath}/**`);
+        childProc.stdout.on('end', () => {
+          result = Buffer.concat(chunks)
+            .toString()
+            .split('\n')
+            .filter((item) => item.length > 0);
+          return resolve(result);
+        });
 
-      if (includePatterns.length) {
-        // at first exclude the whole node_modules directory
-        // after that re-include the production relevant modules
-        exclude = _.union(exclude, [`${pathToDep}node_modules/**`]);
-        include = _.union(include, includePatterns);
+        childProc.stdout.on('error', () => resolve(result));
```

Comment:
Wouldn't it be reasonable to abort in case of error and leave the exclude and includes untouched?